core/web/log_controller.go

```diff
@@ -0,0 +1,101 @@
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/web/presenters"
+	"go.uber.org/zap/zapcore"
+)
+
+// LogController manages the logger config
+type LogController struct {
+	App chainlink.Application
+}
+
+type LoglevelPatchRequest struct {
+	LogLevel string `json:"logLevel"`
+	LogSql   string `json:"logSql"`
+}
+
+func getLogLevelFromStr(logLevel string) (zapcore.Level, error) {
+	switch strings.ToLower(logLevel) {
+	case "debug":
+		return zapcore.DebugLevel, nil
+	case "info":
+		return zapcore.InfoLevel, nil
+	case "warn":
+		return zapcore.WarnLevel, nil
+	case "error":
+		return zapcore.ErrorLevel, nil
+	case "dpanic":
+		return zapcore.DPanicLevel, nil
+	case "panic":
+		return zapcore.PanicLevel, nil
+	case "fatal":
+		return zapcore.FatalLevel, nil
+	default:
+		return zapcore.InfoLevel, fmt.Errorf("could not parse %s as log level (debug, info, warn, error)", logLevel)
+	}
+}
+
+// SetDebug sets the debug log mode for the logger
+func (cc *LogController) SetDebug(c *gin.Context) {
```

Comment:
Rename this to be more consistent with controller function naming
```suggestion
func (cc *LogController) Patch(c *gin.Context) {
```